### DISCUSSION TITLE
Fix for threads on high end Windows systems

### DISCUSSION THREAD

**Discussion by aachrisg** [JuliaLang/julia#44022] [General]
Due to LibUV lacking the necessary pieces, threaded Julia code is only able to take advantage of 1/2 to 1/8 of the available CPU power on Windows systems that either have >64 HW threads (such as the 64-core AMD threadripper), or have more than one CPU socket (such as common Xeon systems).

This fork of LibUV https://github.com/aachrisg/libuv (I submitted a pull request) contains the necessary code in src/win/thread.c to work around this issue. Without this code, my workstation will only show 50% cpu usage now matter how many threads are created.

It is setup to have no affect on systems that wouldn't have the problem.

**Comment by vchuravy** [comment]
x-ref: https://github.com/libuv/libuv/pull/3458
